### DISCUSSION TITLE
Does the reference architecture VPC support S3 endpoints? (In other words, can, say, a "Backend App" connect to S3 despite being in a VPC.)

### DISCUSSION THREAD

**Discussion by marijakstrazdas** [gruntwork-io/knowledge-base#320] [Help] [answered]
Does the reference architecture VPC support S3 endpoints? (In other words, can, say, a "Backend App" connect to S3 despite being in a VPC.)

**Comment by yorinasub17** [comment, marked as answer]
The Reference Architecture deploys the VPC with 2 VPC endpoints: one for S3, and one for DynamoDB. You can further customize it to deploy additional VPC endpoints as you need using our [vpc-interface-endpoint](https://github.com/gruntwork-io/terraform-aws-vpc/tree/master/modules/vpc-interface-endpoint) module, after the hand off process.
